### PR TITLE
Pfe 4216

### PR DESCRIPTION
**Pull request by shane-landry**
## Background
- [x] Adds new slim banner with feature flag
- [x] Adds full width mode with feature flag
- [x] Adds drop shadow toggle feature flag
- [x] Adds close with &times; buttom feature flag
- [x] Adds default background color override 

## Test Plan
```
yarn dev
```
Go to http://localhost:8080/tcf-2.0.html

1. Layer0 - Check "Use Slim"
- [x] Slim banner (layer 0) should load.  
- [x] "Learn More" Should navigate to Layer1.
- [x] "Ok, Got it" should accept all
- [x] Test layout on mobile

2. Full Width Mode - Check "Full Width"
- [x] Layer1 and Vendor list should span full width 
- [x] No rounded corners on Layer1 and Vendor list

3. Toggle Drop Shadow - (Un)Check "Drop Shadow"
- [x] When checked, all layers have drop shadow
- [x] When not checked, no layers have drop shadow

4. Close (&times;) button - Check "Close with &times;"
- [x] When checked, all banner layers should have &times; button displayed in upper right
- [x] When unchecked, not button should render
- [x] Clicking should "accept all" and close CMP UI
- [x] Check positioning on mobile UI

5. Background color override - Set config option theme: { backgroundColor: "#f1f1f1" }
- [x] Banner background colors should display configured color
- [x] No overlapping or nested UI elements should  overlay with another background color

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ See a [working example in codepen](https://codepen.io/potench/pen/GRZZprw).
 			publisherCountryCode: 'US',
 			// language: '', // empty string or unset to default to browser language
 			shouldUseStacks: true,
+			slimMode: true,
```
Comment (potench) [inline README.md]: i would stick with `should, can, has, is` prefix on booleans to stay consistent. 
```
isSlimMode
isFullWidth
```

File: src/s1/components/banner/bannerSlim.jsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+				}}
+			>
+				<div
+					class={[style.content, style.layer1, hasScrolled ? style.scrolling : ''].join(' ')}
+					ref={(el) => (this.scrollRef = el)}
+					style={{
+						maxHeight: maxHeightModal,
+					}}
+				>
+					{ showCloseX && <div class={style.closeX} onClick={this.handleAcceptAll}>&times;</div>}
```
Comment (potench) [inline src/s1/components/banner/bannerSlim.jsx]: checking with product on this, think compliance didn't want close button to accept all originally but that mighta changed

File: README.md (hunk 1 above)
Comment (shane-landry) [reply README.md]: Updated boolean vars / function naming

**Review by potench — APPROVED** [review]

File: src/s1/components/banner/bannerStacks.jsx [hunk 3]
```diff
@@ -174,6 +182,7 @@ export default class BannerStacks extends Component {
 						maxHeight: maxHeightModal,
 					}}
 				>
+					{ shouldShowCloseX && <div class={style.closeX} onClick={this.handleAcceptAll}>&times;</div>}
```
Comment (potench) [inline src/s1/components/banner/bannerStacks.jsx]: can you change this so we can track when user clicks "Close" vs "acceptAll" will be good to see telemetry on that.

```
onClick={this.handleClose}
//  ....

// further up 
handleClose = () => {
    this.handleAcceptAll('acceptAllClose');
}

handleAcceptAll = (clickCategoty = 'acceptAll') => {
 // change logger click category 
};

```

Comment (shane-landry) [reply src/s1/components/banner/bannerStacks.jsx]: Added these to the 3 banners.

**Review by potench — APPROVED** [review]
LGTM

**Pull request merged**